Stratis.Bitcoin/Wallet/Wallet.cs

```diff
@@ -320,8 +357,26 @@ public IEnumerable<TransactionData> GetSpendableTransactions()
         /// <returns></returns>
         public IEnumerable<HdAddress> FindAddressesForTransaction(Func<TransactionData, bool> predicate)
         {
-            var addresses = this.ExternalAddresses.Concat(this.InternalAddresses);
-            return addresses.Where(a => a.Transactions.Any(predicate));
+            Guard.NotNull(predicate, nameof(predicate));
+
+            var addresses = this.GetCombinedAddresses();
+            return addresses.Where(t=> t.Transactions != null).Where(a => a.Transactions.Any(predicate));
+        }
+
+        private IEnumerable<HdAddress> GetCombinedAddresses()
+        {
+            IEnumerable<HdAddress> addresses = new List<HdAddress>();
+            if (this.ExternalAddresses != null)
+            {
+                addresses = addresses.Concat(this.ExternalAddresses);
```

Comment:
I think here you can probably just do addresses = this.ExternalAddresses and concat the internal addresses if != null